netam/models.py

```diff
@@ -642,14 +642,29 @@ def selection_factors_of_aa_str(self, aa_sequence: Tuple[str, str]) -> Tensor:
             )
         idx_seq, mask = self.prepare_aa_str(*aa_sequence)
         with torch.no_grad():
-            result = torch.exp(self.forward(idx_seq, mask))
+            result = torch.exp(self.forward(idx_seq, mask)).squeeze(0)
+        return split_heavy_light_model_outputs(result.cpu(), idx_seq.squeeze(0).cpu())
+
+    def forward(self, amino_acid_indices: Tensor, mask: Tensor) -> Tensor:
+        """Build a binary log selection matrix from an index-encoded parent sequence.
+
+        Because we're predicting log of the selection factor, we don't use an
+        activation function after the transformer.
+
+        Args:
+            amino_acid_indices: A tensor of shape (B, L) containing the indices of parent AA sequences.
+            mask: A tensor of shape (B, L) representing the mask of valid amino acid sites.
+
+        Returns:
+            A tensor of shape (B, L, out_features) representing the log level
+            of selection for each possible amino acid at each site.
+        """
+        result = self.predict(self.represent(amino_acid_indices, mask))
```

Comment:
This seems important and I don't totally get it. Let's talk it over.

Reply:
Oh, I think I get it but still we can chat.

Reply:
This is related to the discussion in the PR thread about how the model makes predictions for the wt amino acid but these predictions should not be used in loss computation. The issue is that wt prediction has been getting applied to synonymous codon probabilities. The fix is to just set the wt prediction to always be 1.